inc/saiport.h

```diff
@@ -2749,6 +2757,68 @@ typedef enum _sai_port_stat_t
     /** Fabric port stat out data units */
     SAI_PORT_STAT_IF_OUT_FABRIC_DATA_UNITS,
 
+    /**
+     * @brief Port FEC codeword symbol error counters.
+     *
+     * This set of counters corresponds to number of errors each FEC codeword
+     * received on the port.
+     *
+     * The number of errors that may be detected is dependent on the type of FEC
+     * in use. For instance, RS-528 FEC supports detection of up to 7 errors,
+     * while RS-544 FEC supports detection of up to 15 errors. The maximum
+     * number of errors that can be detected by the port's current FEC mode may
+     * be retrieved via the SAI_PORT_ATTR_MAX_FEC_ERRORS_DETECTABLE port
+     * attribute.
+     */
+
+    /** Count of FEC codewords with no symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S0,
+
+    /** Count of FEC codewords with 1 symbol error. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S1,
+
+    /** Count of FEC codewords with 2 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S2,
+
+    /** Count of FEC codewords with 3 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S3,
+
+    /** Count of FEC codewords with 4 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S4,
+
+    /** Count of FEC codewords with 5 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S5,
+
+    /** Count of FEC codewords with 6 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S6,
+
+    /** Count of FEC codewords with 7 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S7,
+
+    /** Count of FEC codewords with 8 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S8,
+
+    /** Count of FEC codewords with 9 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S9,
+
+    /** Count of FEC codewords with 10 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S10,
+
+    /** Count of FEC codewords with 11 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S11,
+
+    /** Count of FEC codewords with 12 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S12,
+
+    /** Count of FEC codewords with 13 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S13,
+
+    /** Count of FEC codewords with 14 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S14,
+
+    /** Count of FEC codewords with 15 symbol errors. */
+    SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_S15,
```

Comment:
what will be the value of SAI_PORT_STAT_IF_IN_FEC_CODEWORD_ERRORS_SXX if XX is > max detectable error? 0 ?

Reply:
I've updated the documentation to indicate that if it's greater than max detectable the next bin is used and added an additional counter S16. Let me know what you think.

Reply:
ok